sklearn/utils/estimator_checks.py

```diff
@@ -4502,25 +4503,58 @@ def _output_from_fit_transform(transformer, name, X, df, y):
     return outputs
 
 
-def _check_generated_dataframe(name, case, index, outputs_default, outputs_pandas):
-    import pandas as pd
+def _check_generated_dataframe(
+    name,
+    case,
+    index,
+    outputs_default,
+    outputs_dataframe_lib,
+    is_supported_dataframe,
+    create_dataframe,
+    assert_frame_equal,
+):
+    """Check if the generated DataFrame by the transformer is valid.
+
+    The DataFrame implementation is specified through the parameters of this function.
 
+    Parameters
+    ----------
+    name : str
+        The name of the transformer.
+    case : str
+        A single case from the cases generated by `_output_from_fit_transform`.
+    index : index or None
+        The index of the DataFrame. `None` if the library does not implement a DataFrame
+        with an index.
+    outputs_default : tuple
+        A tuple containing the output data and feature names for the default output.
+    outputs_dataframe_lib : tuple
+        A tuple containing the output data and feature names for the pandas case.
+    is_supported_dataframe : callable
+        A callable that takes a DataFrame instance as input and return whether or
+        E.g. `lambda X: isintance(X, pd.DataFrame)`.
+    create_dataframe : callable
+        A callable taking as parameters `data`, `columns`, and `index` and returns
+        a callable. Be aware that `index` can be ignore if not used.
```

Comment:
Typo:

```suggestion
        a callable. Be aware that `index` can be ignored if not used.
```

although I am not sure about the intended meaning of this statement. "can be ignored if not used" => "should be ignored if not passed"?